src/confcom/azext_confcom/rootfs_proxy.py

```diff
@@ -9,29 +9,62 @@
 import stat
 from pathlib import Path
 import platform
+import requests
 from azext_confcom.errors import eprint
 
 
 host_os = platform.system()
-arch = platform.architecture()[0]
+machine = platform.machine()
 
 
 class SecurityPolicyProxy:  # pylint: disable=too-few-public-methods
     # static variable to cache layer hashes between container groups
     layer_cache = {}
 
+    @staticmethod
+    def download_binaries():
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+
+        bin_folder = os.path.join(dir_path, "bin")
+        if not os.path.exists(bin_folder):
+            os.makedirs(bin_folder)
+
+        # get the most recent release artifacts from github
+        r = requests.get("https://api.github.com/repos/microsoft/hcsshim/releases")
```

Comment:
That request may fail when the user cannot access that url. Please add additional logic to handle that.

Reply:
As mentioned in another comment, this function is only called in setup.py. Would it ever get called on a client machine?